kernel/tests/golden_tables.rs

```diff
@@ -0,0 +1,343 @@
+//! Golden table tests adapted from delta-io/delta tests.
+//!
+//! Data (golden tables) are stored in tests/golden_data/<table_name>.tar.zst
+//! Each table directory has a table/ and expected/ subdirectory with the input/output respectively
+
+use arrow::{compute::filter_record_batch, record_batch::RecordBatch};
+use arrow_array::ArrayRef;
+use arrow_select::concat::concat_batches;
+use paste::paste;
+use std::path::Path;
+use std::sync::Arc;
+
+use arrow_schema::SchemaRef as ArrowSchemaRef;
+use delta_kernel::{engine::arrow_data::ArrowEngineData, DeltaResult, Table};
+use futures::{stream::TryStreamExt, StreamExt};
+use object_store::{local::LocalFileSystem, ObjectStore};
+use parquet::arrow::async_reader::{ParquetObjectReader, ParquetRecordBatchStreamBuilder};
+
+use arrow_array::Array;
+use arrow_schema::DataType;
+use delta_kernel::engine::default::executor::tokio::TokioBackgroundExecutor;
+use delta_kernel::engine::default::DefaultEngine;
+
+#[macro_use]
+mod common;
+use common::to_arrow;
+
+// load the test data from <test_table>.tar.zst file
```

Comment:
nit, make it a doc `///` comment, and: "unpack the test data from test_table.tar.zst into a temp dir, and return the dir it was unpacked into"